main.py

```diff
@@ -96,3 +98,56 @@ def plugin_unloaded():
         if ref_info:
             ref_view.settings().set('refinfo', ref_info.as_value())
     cli.service.exit()
+
+
+_UPDATE_NPM_MESSAGE = "Warning from TypeScript Plugin: We detect a global \
+npm typescript module and it is outdated. Consider upgrading this module by \
+running \"npm install -g typescipt\". TypeScript Plugin still works \
+regardless."
+
+def _check_typescript_version():
+    """
+    Notify user to upgrade npm typescript. Do this only once every time the
+    plugin is updated.
+    """
+    settings = sublime.load_settings('Preferences.sublime-settings')
+    cached_plugin_version = settings.get("typescript_plugin_tsc_version")
+
+    try:
+        plugin_tsc_version = _get_plugin_tsc_version()
+
+        if cached_plugin_version != plugin_tsc_version:
+            # The version number getting from the tsc command is different to
+            # the version number stored in the setting file. This means user
+            # has just updated the plugin.
+
+            npm_tsc_version = _get_npm_tsc_version()
+
+            if npm_tsc_version < plugin_tsc_version:
+                sublime.message_dialog(_UPDATE_NPM_MESSAGE)
```

Comment:
It doesn't necessarily have to be that the plugin version is newer. It may be they are running a newer command-line compiler (e.g. they may have a nightly build or a beta). That could still lead to inconsistencies between the plug-in behavior and 'tsc' on the command line. Just check for 'not equal'